fullscreen.bs

```diff
@@ -392,23 +400,38 @@ could be an open <{dialog}> element.
  <a>active document</a>.
  <!-- cross-process -->
 
+ <li><p>Let <var>pendingDoc</var> be <var>doc</var>.
+
+ <li><p>Let <var>resize</var> be false.
+
  <li><p>If <var>topLevelDoc</var> is in <var>docs</var>, and it is a
- <a>simple fullscreen document</a>, then set <var>doc</var> to <var>topLevelDoc</var> and
+ <a>simple fullscreen document</a>, then set <var>pendingDoc</var> to <var>topLevelDoc</var> and
  <var>resize</var> to true.
 
+ <li><p>Let <var>pending</var> be <var>pendingDoc</var>'s <a>fullscreen element</a>.
+ <!-- cross-process -->
+
  <li><p>Return <var>promise</var>, and run the remaining steps <a>in parallel</a>.
 
- <li><p>If <var>resize</var> is true, resize <var>doc</var>'s viewport to its "normal" dimensions.
+ <li><p>If <var>resize</var> is true, resize <var>pendingDoc</var>'s viewport to its "normal"
+ dimensions.
 
- <li><p>If <var>doc</var>'s <a>fullscreen element</a> is null, then resolve <var>promise</var> with
- undefined and terminate these steps.
+ <li>
+  <p>If <var>pendingDoc</var>'s <a>fullscreen element</a> is not <var>pending</var>:
```

Comment:
@annevk @upsuper, I made another change today in order to unblock this and the tests. Here, if the fullscreen element has changed, just append a pending event for what the fullscreen element was. This will fix the missing event case, but will also mean duplicate events if `document.exitFullscreen()` is called twice or more. It would be possible to avoid this by also having a "fullscreen element removed flag" for "exit fullscreen" that's only set in the removal case, but I'm not sure that's necessary complexity. Things are already pretty weird with multiple enter and exits, and I think that maybe just doing nothing for those cases might be a better fix. Not in this PR, I'm hoping.

Reply:
I have implemented these changes and checked what it does to the tests. /fullscreen/api/document-exit-fullscreen-twice-manual.html and /fullscreen/model/remove-last-manual.html will need some tweaking, but it looks like the idea works.

Reply:
Another way around is to change `list of pending fullscreen events` to just `pending fullscreen event`, which is either null or `(string, element)` pair, and update, rather than append when new event comes.

I may be missing something, but I don't recall why we introduce the list rather than a single slot. Maybe we want to handle `fullscreenerror`? I guess `fullscreenerror` can just be dispatched asynchronously. I see no reason it is necessary to be dispatched in `run fullscreen steps`.

Reply:
That's from https://github.com/whatwg/fullscreen/pull/92. The list is needed in case multiple things happen between two animation frames, in situation like https://github.com/whatwg/fullscreen/issues/63 or if we're in nested fullscreen and call `document.exitFullscreen()` twice. We *could* collapse those into a single `fullscreenchange` event, but that seems a bit odd. We could handle `fullscreenerror` differently, you're right, but that's not really part of the problem as long as the rest remain.

So I think that to do this differently, https://github.com/whatwg/fullscreen/issues/63 and https://github.com/whatwg/fullscreen/issues/119 would have to be resolved in the direction of significant simplification.